pkg/network/dns/snooper_test.go

```diff
@@ -343,13 +336,14 @@ func TestDNSOverNonPort53(t *testing.T) {
 	statKeeper := reverseDNS.statKeeper
 
 	domains := []string{
-		"nonexistent.com.net",
+		"nonexistent.net.com",
 	}
-	h := &handler{}
-	shutdown, port := newTestServer(t, localhost, 0, "udp", h.ServeDNS)
-	defer shutdown()
+	ln, _ := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
For the cases where we need a random port, I think we should just spin up a server and shut it down in the test. 

We can also pass in `0` for the port for the server spin up code to pick a port.

Reply:
that is what we were doing before... I think the idea here was to centralize the server code. but I was on the fence when doing this, wdyt @leeavital

Reply:
I don't have strong feelings, if we prefer to have a server-per-test, I'm fine with it. My thinking was it was nice to have a single global test server because:
* tests run faster when they don't each have to spin up a server
* tests don't have to deal with managing the lifecycle of 
* the nature of a stubbed server makes having a global easy
* (obsolete) it was unlikely for the global server to conflict with other test fixtures since it listened on a nonsense ip (10.10.10.10)

Reply:
I was only proposing a local server for this test where we don't use port 53. The other tests can use the global server. In addition, the server creation/run code seems to be using the same var, `globalServer`, for both servers, which is likely going to be brittle.